datastore/reader/core/reader_service.py

```diff
@@ -1,6 +1,7 @@
 from collections import defaultdict
```

Comment:
All `filter` based routes should also get a span; there, probably a single one suffices for now